crates/red_knot/src/types.rs

```diff
@@ -403,7 +412,13 @@ impl std::fmt::Display for DisplayType<'_> {
 
 #[derive(Debug)]
 pub(crate) struct ClassType {
+    /// Name of the class at definition
     name: Name,
+    /// FileId in which the class was defined
+    pub(crate) file_id: FileId,
```

Comment:
I think it is not redundant due to symbol table being scoped by `FileId`, because the type will travel across modules, and doesn't itself hold a reference to its symbol table.

But I think it is redundant for the reason you noted above, that `ClassTypeId` already carries the `FileId`! So I will remove it.